<a href="https://colab.research.google.com/github/enzoalvino00/Analitica-Empresarial-Integrada/blob/main/Unidad_2_Decision_Experimentaci%C3%B3n%20_Casualidad/Semana_06_Empresa_analitica/Lab_D2/Analitica_Empresarial_Integrada_Lab06.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

---

# 🏛️ **TECSUP 2026-II - ANALÍTICA EMPRESARIAL INTEGRADA**

## 🔬 **LABORATORIO DIRIGIDO N.° 06**: **DISEÑO Y ANÁLISIS DE UN EXPERIMENTO A/B**

| | |
|---|---|
| **Docente** | Pilar Rocío Sayán Mejía |
| **Laboratorio** | Decidir bajo incertidumbre |
| **N.° de laboratorio** | 6 |
| **Tema** | Diseño y análisis de un experimento A/B |
| **Plan curricular** | Big Data y Ciencia de Datos · Código C28 · 2026-I |
| **Duración** | 200 minutos (6:50 p.m. – 10:10 p.m., con receso de 15 minutos) |
| **Modalidad** | Grupal. Las respuestas del foro son individuales. |

---

**Estudiante:** *Enzo Gabriel Alvino Carhuaricra* &nbsp;&nbsp;&nbsp; **Sección:** **6C28** &nbsp;&nbsp;&nbsp; **Fecha:** 23/09/2026

### **Pregunta orientadora**

**¿Qué evidencia estadística y práctica necesitamos antes de implementar una variante B?**

> Este notebook acompaña el trabajo desarrollado durante la clase. No es un informe ni un proyecto para exponer. Las celdas alternan explicación breve, ejecución guiada, interpretación y retos de cierre.


# I. Elemento de la capacidad terminal de la semana

Diseña y analiza un experimento A/B identificando el factor, sus niveles, el control, el tratamiento, la unidad experimental y la variable respuesta. Formula las hipótesis estadísticas, establece la métrica primaria, los guardrails, el nivel de significación, la potencia y el efecto mínimo detectable; finalmente, integra la evidencia estadística con la relevancia práctica para sustentar una decisión.


# II. Seguridad y condiciones de trabajo

- Utilice únicamente Google Colab y las fuentes indicadas por la docente.
- No modifique ni fabrique resultados para forzar una conclusión.
- Ejecute las celdas en el orden propuesto y documente cualquier error o dato ausente.
- No comparta claves, credenciales ni información personal en el notebook.
- Mantenga la estación de trabajo ordenada y no instale software no autorizado.


# III. Fundamento teórico
Un A/B test compara un control A con una variante B mediante asignación aleatoria. La unidad experimental de este caso es el usuario, que conserva su variante durante toda la observación. La aleatorización favorece la comparabilidad en expectativa; no garantiza grupos idénticos.

La respuesta es binaria: compra dentro de los siete días posteriores a la asignación. Todos los usuarios completan esa ventana antes del análisis. La métrica primaria es la proporción de usuarios que compran. Se estudia la diferencia de dos proporciones, no ANOVA.

El MDE es el efecto utilizado para planificar la potencia. El umbral empresarial es la mejora que justificaría interés práctico. Aquí son distintos: MDE de 1,5 pp y umbral empresarial de 1,0 pp. La muestra se dimensiona para detectar el MDE frente a cero, no para garantizar evidencia de superar el umbral empresarial.

Se usa una aproximación normal de Wald para la diferencia. El contraste unilateral al 5 % y el límite inferior unilateral del 95 % utilizan el mismo error estándar. También se muestra un IC bilateral del 95 % como descripción de incertidumbre; no se exige que este último excluya cero para reproducir el contraste unilateral.

La recomendación académica requiere evidencia de superar el umbral empresarial y de mantener el guardrail dentro del margen predefinido. Son criterios más exigentes que obtener p < 0,05 frente a cero. Un resultado no concluyente no demuestra igualdad.


# IV. Normas y buenas prácticas empleadas

- Prerregistro lógico: definir hipótesis, métrica, MDE y regla de decisión antes de observar los resultados.
- Aleatorización: evitar que la asignación dependa de características de las unidades.
- Trazabilidad: conservar la fuente, las transformaciones y los parámetros utilizados.
- Reproducibilidad: ejecutar el análisis mediante código verificable.
- No realizar *peeking*: no detener o prolongar el experimento solo porque el resultado momentáneo conviene.
- No contaminar grupos: una unidad debe recibir únicamente la variante asignada.
- Distinguir asociación, evidencia experimental y decisión práctica.


# V. Recursos
- Computadora, Google Colab y notebook LAB-D6-AEI-PSAYAN-2026-02.ipynb.
- Librerías numpy, pandas, scipy, statsmodels y plotly.
- Caso ficticio NexaCommerce con datos simulados generados en el propio notebook. No requiere descargar un dataset externo.

## Naturaleza de los datos utilizados
Los registros son sintéticos y reproducibles mediante una semilla fija. Representan usuarios de un checkout; no proceden de una empresa real. Las probabilidades del simulador son supuestos académicos, no evidencia observada. No cambie la semilla para buscar un resultado favorable. En un estudio real, los datos procederían de registros del experimento y el plan se documentaría antes de recogerlos.


# VI. Metodología para el desarrollo de la sesión

El laboratorio se desarrolla mediante una secuencia guiada de cinco momentos:

1. **Comprender:** recuperar el vocabulario esencial.
2. **Representar:** traducir el caso a la estructura de un diseño experimental.
3. **Formular:** establecer las hipótesis y la regla de decisión antes de observar resultados.
4. **Ejecutar e interpretar:** correr el código por etapas y explicar cada salida.
5. **Decidir y transferir:** integrar evidencia estadística, umbral empresarial y guardrails; luego resolver retos breves.

No se solicita exposición, informe ni trabajo adicional. La evidencia de aprendizaje se construye durante la clase mediante respuestas breves en las pausas de análisis.


# VII. Procedimiento

## Actividad 1 — Activación conceptual

**Propósito:** reconocer los componentes que deben definirse antes de ejecutar un experimento.

Complete la tabla con sus propias palabras. La pregunta de apoyo permite verificar si la definición es suficiente.

| Concepto | Definición con sus propias palabras | Pregunta de apoyo |
|---|---|---|
| **Experimento A/B** | Es una prueba controlada donde se comparan dos versiones, A y B, mediante asignación aleatoria para evaluar si un cambio produce un efecto. | ¿Qué diferencia existe entre comparar dos grupos y experimentar? |
| **Control** | Es la versión actual o de referencia que sirve como base para comparar los resultados. | ¿Qué representa el punto de referencia? |
| **Tratamiento** | Es la nueva versión o cambio que se quiere probar en el experimento. | ¿Qué cambio recibe este grupo? |
| **Unidad experimental** | Es la persona, usuario o elemento sobre el que se aplica una de las variantes del experimento. | ¿Sobre quién o qué se aplica la variante? |
| **Variable respuesta** | Es el resultado que se observa en cada unidad experimental después de recibir una variante. | ¿Qué resultado se observa en cada unidad? |
| **Métrica primaria** | Es la medida principal que permite evaluar si se alcanzó el objetivo del experimento. | ¿Cuál será el criterio principal de éxito? |
| **Guardrail** | Es una métrica que no debe deteriorarse aunque la métrica primaria mejore. | ¿Qué no debería deteriorarse? |
| **MDE** | Es el efecto mínimo que se utiliza para planificar la potencia y determinar el tamaño de muestra necesario. | ¿Qué efecto se utiliza para planificar la potencia? |
| **Potencia** | Es la capacidad que tiene el experimento para detectar un efecto real del tamaño definido por el MDE. | ¿Qué capacidad tiene el diseño para detectar el MDE? |
| **Peeking** | Es revisar los resultados parciales y tomar decisiones antes de que termine el experimento según lo observado. | ¿Por qué no se debe decidir mirando resultados parciales? |

### Pausa de verificación

**¿Por qué el MDE no debe elegirse después de conocer la diferencia observada?**

El MDE debe definirse antes de ejecutar el experimento porque se utiliza para planificar la potencia y el tamaño de muestra. Elegirlo después de observar los resultados adaptaría el criterio a los datos obtenidos y afectaría la planificación previa del experimento.

En este laboratorio:

- **MDE = 1,5 puntos porcentuales:** se utiliza para planificar la potencia y el tamaño de muestra.
- **Umbral empresarial = 1,0 punto porcentual:** se utiliza para evaluar si la mejora tiene relevancia práctica.


## Actividad 2 — Interpretación y representación metodológica
NexaCommerce, empresa ficticia, compara el checkout actual de cuatro pasos con uno simplificado de dos pasos. Cada usuario recibe una sola variante. Se mide compra e incidencia de pago dentro de siete días; incluso quienes no compran permanecen en el denominador.

### 2.1 Interpretación
Complete antes de ejecutar: Determinar si **el checkout simplificado de dos pasos incrementa la proporción de usuarios que compran dentro de siete días sin deteriorar la incidencia de pago más allá de +0,5 puntos porcentuales**.

### 2.2 Representación del diseño
| Elemento | Definición en este experimento |
|---|---|
| Factor y niveles | Checkout: A actual; B simplificado |
| Unidad experimental | Usuario único con asignación persistente |
| Respuesta primaria | compra_7d: 1 si compró dentro de siete días; 0 si no |
| Métrica primaria | Usuarios que compraron / todos los usuarios asignados |
| Guardrail | incidencia_pago_7d: al menos una incidencia por usuario en siete días |
| Denominador del guardrail | Todos los usuarios asignados, compren o no |
| Margen de deterioro | B−A debe ser inferior a +0,5 pp en incidencias |
| Baseline histórico supuesto | Conversión 11,8 %; incidencias 1,8 % |
| MDE de conversión | +1,5 pp para planificar potencia de 80 % frente a cero |
| Umbral empresarial | Mejora de conversión superior a +1,0 pp |
| Horizonte | Reclutamiento planificado de 14 días y 7 días de maduración |
| Muestra | Se calcula antes de simular las respuestas |

### 2.3 Hipótesis y reglas previas
Conversión: H0: pB−pA ≤ 0; H1: pB−pA > 0. Alfa = 0,05 unilateral.

Relevancia empresarial: exigir límite inferior unilateral del 95 % mayor que 0,010. Esto evalúa un umbral distinto de cero y no se deduce solo de la estimación puntual.

Guardrail: H0: qB−qA ≥ 0,005; H1: qB−qA < 0,005. Exigir límite superior unilateral del 95 % menor que 0,005. No encontrar una diferencia significativa no demuestra que el guardrail esté protegido.

La recomendación combina ambos requisitos; se trata de una regla conjunta, no de escoger la prueba favorable. Los intervalos individuales no deben describirse como una región de confianza simultánea.

**Comprobación:** ¿por qué se comparan proporciones? ¿Cómo cambiarían las hipótesis si interesara cualquier diferencia, positiva o negativa?
**Respuesta:** Se comparan proporciones porque la variable respuesta es binaria y la métrica primaria representa la proporción de usuarios que compran. Si interesara cualquier diferencia, positiva o negativa, la hipótesis alternativa sería bilateral: H1: pB−pA ≠ 0, en lugar de H1: pB−pA > 0.


## Actividad 3 — Preparación y planificación previa
### Paso 1. Preparar Google Colab
Ejecute las celdas en orden. La configuración y el cálculo de muestra aparecen antes de generar las respuestas. En Colab, la primera celda instala solo los paquetes que faltan. Si se solicita reiniciar, vuelva a ejecutar desde aquí.


In [ ]:
# ============================================================
# CONFIGURACIÓN DEL ENTORNO Y DEPENDENCIAS
# ============================================================

# Importar herramientas necesarias para verificar e instalar paquetes
import importlib.util
import subprocess
import sys

# Definir los paquetes necesarios para ejecutar el laboratorio
paquetes = ["numpy", "pandas", "scipy", "statsmodels", "plotly"]

## Identificar los paquetes que aún no están instalados
faltan = [p for p in paquetes if importlib.util.find_spec(p) is None]

# Instalar únicamente los paquetes faltantes
if faltan:
    ## Utilizar el mismo intérprete de Python que ejecuta el notebook
    subprocess.check_call([sys.executable, "-m", "pip", "install", "-q", *faltan])

# Confirmar que las dependencias necesarias están disponibles
print("Dependencias disponibles.")


Dependencias disponibles.


In [ ]:
# ============================================================
# IMPORTACIÓN Y CONFIGURACIÓN DE LIBRERÍAS
# ============================================================

# Importar NumPy para cálculos numéricos y manejo de datos
import numpy as np

# Importar pandas para trabajar con tablas y estructuras de datos
import pandas as pd

# Importar Plotly Express para generar visualizaciones
import plotly.express as px

# Importar la distribución normal para cálculos estadísticos
from scipy.stats import norm

# Importar la herramienta para calcular la potencia estadística
from statsmodels.stats.power import NormalIndPower

# Importar la función para calcular el tamaño del efecto entre proporciones
from statsmodels.stats.proportion import proportion_effectsize

# Configurar pandas para mostrar hasta 20 columnas en las tablas
pd.set_option("display.max_columns", 20)

# Confirmar que el entorno de trabajo está listo
print("Entorno listo.")


Entorno listo.


### Paso 2. Fijar los parámetros antes de observar
Los baselines son supuestos históricos del caso ficticio, no estimaciones obtenidas de los grupos que analizaremos. El margen del guardrail se fija por criterio de negocio. Se calcula muestra para conversión y, por aproximación normal, para no inferioridad del guardrail suponiendo diferencia verdadera cero. Se toma el mayor tamaño. Las potencias se refieren a cada objetivo de diseño; no garantizan 80 % para la regla conjunta ni para superar el umbral empresarial.


In [ ]:
# ============================================================
# CONFIGURACIÓN DE PARÁMETROS DEL EXPERIMENTO
# ============================================================

# Definir el nivel de significancia y la potencia estadística
ALPHA = 0.05  ### Nivel de significancia del 5 %
POTENCIA = 0.80  ### Potencia estadística del 80 %

# Definir el baseline de conversión y el efecto mínimo detectable
BASELINE = 0.118  ### Conversión histórica del caso: 11.8 %
MDE = 0.015  ### Efecto mínimo detectable: 1.5 puntos porcentuales

# Definir el umbral de negocio
UMBRAL_NEGOCIO = 0.010  ### Aumento mínimo considerado relevante para el negocio: 1.0 punto porcentual

# Definir el baseline y el margen permitido para el guardrail
BASE_GUARDRAIL = 0.018  ### Incidencia histórica del guardrail: 1.8 %
MARGEN_GUARDRAIL = 0.005  ### Deterioro máximo permitido: 0.5 puntos porcentuales

# Fijar la semilla para garantizar la reproducibilidad
SEMILLA = 20260921  ### Semilla utilizada para obtener los mismos resultados en la simulación

# Definir el periodo de reclutamiento y la ventana de seguimiento
DIAS_RECLUTAMIENTO = 14  ### Días destinados al reclutamiento de usuarios
VENTANA_DIAS = 7  ### Días de seguimiento para medir la compra

# ============================================================
# VALIDACIÓN DE LOS PARÁMETROS
# ============================================================

# Verificar que el baseline y el MDE produzcan una proporción válida
assert 0 < BASELINE < BASELINE + MDE < 1  ### Comprueba que las proporciones estén entre 0 y 1

# Verificar que el umbral de negocio sea positivo y menor que el MDE
assert 0 < UMBRAL_NEGOCIO < MDE  ### Mantiene diferenciados el criterio estadístico y el criterio empresarial

# ============================================================
# CÁLCULO DEL TAMAÑO DE MUESTRA PARA CONVERSIÓN
# ============================================================

# Crear el objeto para calcular potencia y tamaño de muestra
analysis = NormalIndPower()

# Calcular el tamaño del efecto entre la conversión base y la conversión esperada
es = proportion_effectsize(BASELINE + MDE, BASELINE)  ### Compara 11.8 % frente a 13.3 %

# Calcular la muestra necesaria para detectar el MDE con la potencia establecida
n_conversion = int(np.ceil(analysis.solve_power(
    effect_size=es, alpha=ALPHA, power=POTENCIA,
    ratio=1, alternative="larger")))  ### Prueba unilateral para detectar un aumento de B sobre A

# ============================================================
# CÁLCULO DEL TAMAÑO DE MUESTRA PARA EL GUARDRAIL
# ============================================================

# Obtener los valores críticos de la distribución normal
z_alpha = norm.ppf(1 - ALPHA)  ### Valor crítico asociado al nivel de significancia
z_power = norm.ppf(POTENCIA)  ### Valor asociado a la potencia estadística

# Calcular la muestra aproximada necesaria para evaluar el margen del guardrail
n_guardrail = int(np.ceil(
    2 * BASE_GUARDRAIL * (1 - BASE_GUARDRAIL)
    * (z_alpha + z_power)**2 / MARGEN_GUARDRAIL**2))  ### Se calcula para evaluar el margen de +0.5 pp del guardrail

# ============================================================
# DEFINICIÓN DE LA MUESTRA PLANIFICADA
# ============================================================

# Seleccionar el mayor tamaño de muestra entre conversión y guardrail
N_POR_GRUPO = max(n_conversion, n_guardrail)  ### Se toma el mayor para cubrir ambos objetivos de diseño

# Mostrar los tamaños de muestra calculados
print("Muestra por grupo para conversión:", n_conversion)
print("Muestra aproximada por grupo para guardrail:", n_guardrail)
print("Muestra planificada por grupo:", N_POR_GRUPO)

# Indicar que el análisis debe realizarse una sola vez al completar el experimento
print("Analizar una sola vez al completar muestra y seguimiento.")  ### Evita el peeking antes de completar muestra y seguimiento

Muestra por grupo para conversión: 6027
Muestra aproximada por grupo para guardrail: 8743
Muestra planificada por grupo: 8743
Analizar una sola vez al completar muestra y seguimiento.


### **Tamaño de muestra planificado**

Se necesitan **6027 usuarios por grupo** para detectar el MDE de **1.5 puntos porcentuales** en la conversión. Para el guardrail se requieren **8743 usuarios por grupo**, por lo que se toma este valor como muestra final.

En total, el experimento necesitará **17 486 usuarios**, distribuidos entre A y B. Además, el análisis debe realizarse **una sola vez después de completar la muestra y el seguimiento de 7 días**, evitando revisar resultados anticipadamente.


### **Pausa de análisis 1**

1. ¿Por qué el baseline debe establecerse antes de ver el resultado?

Porque el baseline debe definirse antes de observar los resultados para evitar adaptar el criterio de análisis al resultado obtenido.

2. ¿Por qué la muestra final considera también el guardrail?

Porque el experimento debe contar con suficiente muestra para evaluar tanto la métrica primaria como el guardrail y asegurar que ambos objetivos del diseño puedan ser analizados.

3. ¿La potencia calculada garantiza demostrar una mejora superior a 1,0 pp?
No. La potencia fue calculada para detectar el **MDE de 1,5 pp frente a cero**, por lo que no garantiza demostrar una mejora superior al **umbral empresarial de 1,0 pp**.


### Paso 3. Generar y validar el conjunto académico
El simulador asigna aleatoriamente el mismo número de usuarios a A y B. Genera una fila por usuario y completa siete días de observación para todos. Supone conversiones de 11,8 % y 13,8 % e incidencias de 1,8 % en ambos grupos. Son parámetros del generador, no resultados que deban recuperarse exactamente. Las respuestas se simulan independientes para simplificar el ejercicio; una empresa real puede presentar dependencias y estacionalidad no representadas aquí.


In [ ]:
# ============================================================
# GENERACIÓN DEL CONJUNTO DE DATOS SIMULADO
# ============================================================

# Crear el generador de números aleatorios utilizando la semilla definida
rng = np.random.default_rng(SEMILLA)  ### Permite obtener resultados reproducibles en la simulación

# Calcular el número total de usuarios del experimento
total = 2 * N_POR_GRUPO  ### Se consideran dos grupos: A_Control y B_Tratamiento

# Asignar aleatoriamente el mismo número de usuarios a cada grupo
grupos = rng.permutation(np.repeat(["A_Control", "B_Tratamiento"], N_POR_GRUPO))  ### La permutación evita un orden fijo de asignación

# Definir la probabilidad de compra según el grupo asignado
prob_compra = np.where(grupos == "A_Control", 0.118, 0.138)  ### A tiene 11.8 % y B 13.8 % como parámetros del simulador

# ============================================================
# CONSTRUCCIÓN DE LA TABLA DE USUARIOS
# ============================================================

# Crear una fila por cada usuario con sus variables del experimento
df = pd.DataFrame({

    "usuario_id": np.arange(1, total + 1),  ### Generar un identificador único para cada usuario

    "grupo": grupos,  ### Registrar la variante asignada a cada usuario

    "dia_asignacion": rng.integers(1, DIAS_RECLUTAMIENTO + 1, total),  ### Asignar aleatoriamente el día de ingreso al experimento

    "dias_observados": np.full(total, VENTANA_DIAS),  ### Completar siete días de observación para todos los usuarios

    "compra_7d": rng.binomial(1, prob_compra),  ### Simular si cada usuario realizó una compra dentro de los siete días

    "incidencia_pago_7d": rng.binomial(1, BASE_GUARDRAIL, total)  ### Simular la incidencia de pago utilizando el baseline de 1.8 %
})

# ============================================================
# VALIDACIÓN INICIAL DEL CONJUNTO SIMULADO
# ============================================================

# Mostrar información básica del conjunto de datos generado
print("DATOS SIMULADOS — NexaCommerce")

# Mostrar la cantidad de filas y columnas del conjunto de datos
print("Dimensión:", df.shape)  ### Permite verificar que se generó una fila por usuario y las variables esperadas

# Mostrar las primeras filas para revisar la estructura de los datos
display(df.head())


DATOS SIMULADOS — NexaCommerce
Dimensión: (17486, 6)


,usuario_id,grupo,dia_asignacion,dias_observados,compra_7d,incidencia_pago_7d
0,1,A_Control,8,7,0,0
1,2,B_Tratamiento,5,7,0,0
2,3,A_Control,12,7,0,0
3,4,A_Control,9,7,0,0
4,5,B_Tratamiento,12,7,0,0


### **Validación del conjunto de datos simulado**

El conjunto generado contiene **17 486 usuarios y 6 variables**, lo que coincide con la muestra planificada de **8743 usuarios por grupo**. Cada usuario tiene un identificador, grupo asignado, día de asignación, siete días de observación, resultado de compra e incidencia de pago.

La tabla muestra que los usuarios fueron asignados a **A_Control o B_Tratamiento** y que todos cuentan con los **7 días de observación**, cumpliendo las condiciones establecidas para el experimento.



In [ ]:
# ============================================================
# VALIDACIÓN DE LA CALIDAD Y ESTRUCTURA DE LOS DATOS
# ============================================================

# Definir las columnas indispensables para el análisis
requeridas = {"usuario_id", "grupo", "dias_observados",
              "compra_7d", "incidencia_pago_7d"}

# Verificar que todas las columnas necesarias estén presentes
assert requeridas.issubset(df.columns), "Faltan columnas"

# Verificar que no existan valores ausentes en las variables principales
assert not df[list(requeridas)].isna().any().any(), "Hay datos ausentes"  ### Cada usuario debe tener una observación completa

# Verificar que cada usuario aparezca una sola vez
assert df["usuario_id"].is_unique, "Usuarios repetidos o contaminados"  ### Evita que un usuario aparezca más de una vez

# Verificar que únicamente existan los dos grupos definidos
assert set(df["grupo"]) == {"A_Control", "B_Tratamiento"}

# Verificar que todos los usuarios tengan completa la ventana de seguimiento
assert (df["dias_observados"] == VENTANA_DIAS).all()  ### Todos deben tener los 7 días de observación

# Verificar que las variables de respuesta sean binarias
for col in ["compra_7d", "incidencia_pago_7d"]:
    assert df[col].isin([0, 1]).all(), f"{col} no es binaria"  ### 1 indica que ocurrió el evento y 0 que no ocurrió

# Crear una copia del conjunto validado para continuar con el análisis
ab = df.copy()

# Calcular la cantidad de usuarios asignados a cada grupo
tam_grupos = ab.groupby("grupo").size()

# Verificar que ambos grupos tengan exactamente el tamaño planificado
assert (tam_grupos == N_POR_GRUPO).all()  ### Cada grupo debe contener 8743 usuarios

# Mostrar el número de usuarios por grupo
display(tam_grupos.to_frame("usuarios"))

# Confirmar que las validaciones fueron superadas
print("Calidad estructural y seguimiento completos.")


,usuarios
grupo,
A_Control,8743
B_Tratamiento,8743


Calidad estructural y seguimiento completos.


### **Validación de los grupos experimentales**

La validación confirma que ambos grupos tienen exactamente **8743 usuarios**, cumpliendo con el tamaño de muestra planificado. El mensaje **“Calidad estructural y seguimiento completos”** indica que las verificaciones realizadas sobre la estructura de los datos y el seguimiento fueron superadas correctamente.

Por lo tanto, el conjunto de datos queda preparado para continuar con el **análisis de la métrica primaria y del guardrail**.

### **Pausa de análisis 2**

1. Registre el tamaño de cada grupo. ¿Cumple el plan?

**Respuesta:** A_Control tiene **8743 usuarios** y B_Tratamiento tiene **8743 usuarios**. Sí, ambos grupos cumplen con el tamaño de muestra planificado.

2. ¿Qué ocurriría si un usuario apareciera en ambas variantes?

**Respuesta:** Se produciría **contaminación del experimento**, porque no podríamos atribuir correctamente el resultado a una sola variante y se afectaría la comparación entre A y B.

3. ¿Por qué no debe excluirse a quienes no compraron?

**Respuesta:** Porque la métrica primaria es la **proporción de usuarios que compran sobre todos los usuarios asignados**. Excluir a quienes no compraron alteraría el denominador y sesgaría la medición de conversión.

4. ¿Por qué es necesario completar los siete días antes de analizar?

**Respuesta:** Porque la compra se mide dentro de una **ventana de siete días**. Esperar a completar ese periodo permite que todos los usuarios tengan la misma oportunidad de registrar una compra antes de comparar A y B.


## Actividad 4 — MDE, potencia y sensibilidad de muestra
### Paso 4. Recuperar el diseño fijado
Revise los parámetros sin recalibrarlos con el resultado observado. El MDE sirve para dimensionar la muestra; el umbral empresarial se utiliza para valorar la mejora. En este caso, las dos cantidades difieren deliberadamente.


In [ ]:
# ============================================================
# RECUPERACIÓN DE LOS PARÁMETROS DEL DISEÑO
# ============================================================

# Mostrar los principales parámetros definidos antes de observar los resultados
display(pd.DataFrame({
    "parametro": ["Baseline", "MDE", "Umbral empresarial",
                  "Margen guardrail", "Alfa", "Potencia"],
    "valor": [BASELINE, MDE, UMBRAL_NEGOCIO,
              MARGEN_GUARDRAIL, ALPHA, POTENCIA]
}))

# Mostrar la muestra final planificada para cada grupo
print("Muestra final por grupo:", N_POR_GRUPO)  ### Corresponde al mayor tamaño requerido entre conversión y guardrail


,parametro,valor
0,Baseline,0.118
1,MDE,0.015
2,Umbral empresarial,0.010
3,Margen guardrail,0.005
4,Alfa,0.050
5,Potencia,0.800


Muestra final por grupo: 8743


### **Recuperación de los parámetros del diseño**

Los parámetros se mantienen sin modificaciones después de generar los datos: **baseline = 11.8 %**, **MDE = 1.5 pp**, **umbral empresarial = 1.0 pp**, **margen del guardrail = 0.5 pp**, **alfa = 5 %** y **potencia = 80 %**. La muestra final queda en **8743 usuarios por grupo**, porque este fue el mayor tamaño requerido entre la métrica de conversión y el guardrail.

La diferencia entre el **MDE de 1.5 pp** y el **umbral empresarial de 1.0 pp** es intencional: el primero se utiliza para dimensionar la muestra y el segundo para valorar la relevancia de la mejora.

### Paso 5. Explorar la sensibilidad al MDE
Compare MDE de 0,5, 1,5 y 2,5 pp. Esta exploración es didáctica y no cambia el plan ni la decisión del caso ya fijado.
**Predicción:** ¿qué efecto necesitará más muestra y por qué?


In [ ]:
# ============================================================
# ANÁLISIS DE SENSIBILIDAD AL MDE
# ============================================================

# Crear una lista para almacenar los resultados de cada escenario
filas = []

# Evaluar diferentes valores de MDE
for mde in [0.005, 0.015, 0.025]:

    # Calcular el tamaño del efecto para cada MDE
    es = proportion_effectsize(BASELINE + mde, BASELINE)  ### Compara el baseline con el aumento definido por cada MDE

    # Calcular la muestra necesaria para detectar cada MDE
    n = analysis.solve_power(effect_size=es, alpha=ALPHA,
                             power=POTENCIA, alternative="larger")  ### Mantiene alfa, potencia y prueba unilateral constantes

    # Guardar el MDE y los tamaños de muestra calculados
    filas.append({"MDE_pp": mde * 100,
                  "n_conversion": int(np.ceil(n)),  ### Redondear hacia arriba la muestra necesaria para conversión
                  "n_con_guardrail": max(int(np.ceil(n)), n_guardrail)})  ### Tomar el mayor tamaño entre conversión y guardrail

# Convertir los resultados en una tabla
sensibilidad_n = pd.DataFrame(filas)

# Mostrar la comparación de tamaños de muestra
display(sensibilidad_n)

# ============================================================
# VISUALIZACIÓN DE LA SENSIBILIDAD
# ============================================================

# Crear una gráfica para observar cómo cambia la muestra según el MDE
fig = px.line(sensibilidad_n, x="MDE_pp",
              y=["n_conversion", "n_con_guardrail"], markers=True,
              title="Muestra por grupo según el MDE",
              labels={"value": "Usuarios por grupo", "MDE_pp": "MDE (pp)"})

# Mostrar la gráfica
fig.show()

,MDE_pp,n_conversion,n_con_guardrail
0,0.5,52413,52413
1,1.5,6027,8743
2,2.5,2241,8743


### **Sensibilidad del tamaño de muestra al MDE**

Los resultados confirman la predicción: cuando el **MDE disminuye**, la muestra necesaria aumenta considerablemente. Con un MDE de **0.5 pp** se requieren **52 413 usuarios por grupo**, mientras que con **1.5 pp** se necesitan **6027** y con **2.5 pp**, solo **2241** para la métrica de conversión.

Sin embargo, al considerar el guardrail, la muestra final es **52 413 usuarios por grupo** para MDE de 0.5 pp y **8743 usuarios por grupo** tanto para 1.5 como para 2.5 pp. Esto muestra que un efecto más pequeño requiere mucha más información para poder detectarse con la misma potencia estadística.

### **Pausa de análisis 3**

1. Explique por qué reducir el MDE aumenta la muestra para conversión.

Porque un MDE más pequeño representa un efecto más difícil de detectar frente a la variabilidad aleatoria. Por eso se necesita una mayor cantidad de usuarios para mantener la misma potencia estadística.

2. ¿Por qué el guardrail puede determinar la muestra final?

Porque la muestra final debe ser suficiente para cumplir tanto el objetivo de la métrica primaria como el del guardrail. Por ello, se toma el mayor tamaño de muestra requerido entre ambos.

3. ¿Sería válido aumentar el MDE después de ver los resultados para declarar éxito?

No. El MDE debe fijarse **antes de observar los resultados**, porque cambiarlo después permitiría adaptar el criterio de evaluación a los datos obtenidos y afectaría la validez del análisis.


## Actividad 5 — Análisis estadístico de la comparación A/B

**Propósito:** calcular el efecto observado y evaluar la hipótesis mediante una prueba para dos proporciones.

### Paso 6. Obtener tasas y tamaño del efecto observado

Antes de ejecutar, recuerde:

- Diferencia absoluta: $p_B-p_A$.
- Lift relativo: $(p_B-p_A)/p_A$.
- La diferencia se expresa en puntos porcentuales; el lift expresa un cambio relativo respecto de A.


In [ ]:
# ============================================================
# CÁLCULO DE TASAS Y EFECTO OBSERVADO
# ============================================================

# Calcular compras, tamaño de muestra y tasa de conversión por grupo
tabla = ab.groupby("grupo")["compra_7d"].agg(["sum", "count", "mean"])

# Renombrar las columnas para facilitar su interpretación
tabla.columns = ["compras", "n", "tasa_conversion"]

# Obtener el número de compras y usuarios del grupo de control
x_A, n_A = int(tabla.loc["A_Control", "compras"]), int(tabla.loc["A_Control", "n"])  ### x_A representa compras y n_A el total de usuarios de A

# Obtener el número de compras y usuarios del grupo de tratamiento
x_B, n_B = int(tabla.loc["B_Tratamiento", "compras"]), int(tabla.loc["B_Tratamiento", "n"])  ### x_B representa compras y n_B el total de usuarios de B

# Calcular las tasas de conversión de ambos grupos
p_A, p_B = x_A / n_A, x_B / n_B  ### Cada tasa corresponde a compras divididas entre todos los usuarios del grupo

# Calcular la diferencia absoluta de conversión entre B y A
efecto = p_B - p_A  ### Representa el cambio observado en puntos porcentuales

# Calcular el lift relativo respecto al grupo A
lift_rel = efecto / p_A if p_A > 0 else np.nan  ### Mide el cambio porcentual de B respecto a la conversión de A

# Mostrar la tabla de resultados por grupo
display(tabla)

# Mostrar la diferencia absoluta entre las conversiones
print(f"Diferencia B−A: {efecto*100:.3f} pp")

# Mostrar el cambio relativo de B respecto a A
print(f"Lift relativo: {lift_rel:.2%}")

,compras,n,tasa_conversion
grupo,,,
A_Control,996,8743,0.11392
B_Tratamiento,1236,8743,0.14137


Diferencia B−A: 2.745 pp
Lift relativo: 24.10%


### **Tasas de conversión y efecto observado**

El grupo **A_Control** obtuvo una tasa de conversión de **11.392 %**, mientras que **B_Tratamiento** alcanzó **14.137 %**. La diferencia observada entre ambos grupos es de **+2.745 puntos porcentuales**.

El **lift relativo de 24.10 %** indica que la conversión de B fue aproximadamente 24.10 % mayor que la de A en términos relativos. Estos resultados describen el efecto observado en los datos; todavía es necesario realizar la prueba estadística para determinar si existe evidencia suficiente de una diferencia entre los grupos.

### **Pausa de análisis 4**

> La tasa de B es **14.137 %** y la tasa de A es **11.392 %**. La diferencia observada es **2.745 puntos porcentuales**, equivalente a un lift relativo de **24.10 %**. Estos valores describen la muestra, pero todavía no permiten afirmar que **la diferencia observada sea estadísticamente significativa**.


### Paso 7. Contraste e intervalos coherentes
Usamos la aproximación normal de Wald con error estándar no combinado. Se comprueba que haya al menos diez compras y diez no compras en cada grupo; esta regla práctica no sustituye la revisión del diseño.

Para la prueba unilateral frente a cero, p < 0,05 equivale a límite inferior unilateral del 95 % > 0 con el mismo método. El IC bilateral del 95 % se presenta aparte. La función también permite contrastar un umbral distinto de cero. El p-valor no es la probabilidad de que H0 sea verdadera.


In [ ]:
# ============================================================
# CONTRASTE DE WALD E INTERVALOS DE CONFIANZA
# ============================================================
def diferencia_wald(x_b, n_b, x_a, n_a, umbral=0.0):

    # Verificación de frecuencias mínimas
    if min(x_b, n_b-x_b, x_a, n_a-x_a) < 10:
        raise ValueError("Frecuencias pequeñas: revisar método antes de continuar")

    pb, pa = x_b/n_b, x_a/n_a  ### Calcula la tasa de conversión observada en cada grupo.

    d = pb-pa  ### Calcula la diferencia absoluta entre B y A.

    se = np.sqrt(pb*(1-pb)/n_b + pa*(1-pa)/n_a)  ### Obtiene el error estándar no combinado de la diferencia.

    z = (d-umbral)/se  ### Estandariza la diferencia respecto al umbral que se desea contrastar.

    return {"d": d, "se": se, "z": z,
            "p_mayor": norm.sf(z),  ### Calcula el p-valor para una alternativa de que B sea mayor que A.
            "p_menor": norm.cdf(z),  ### Calcula el p-valor para una alternativa de que B sea menor que A.
            "li_uni95": d-norm.ppf(1-ALPHA)*se,  ### Calcula el límite inferior unilateral del 95 %.
            "ls_uni95": d+norm.ppf(1-ALPHA)*se,  ### Calcula el límite superior unilateral del 95 %.
            "ic95_low": d-norm.ppf(1-ALPHA/2)*se,  ### Calcula el límite inferior del IC bilateral del 95 %.
            "ic95_high": d+norm.ppf(1-ALPHA/2)*se}  ### Calcula el límite superior del IC bilateral del 95 %.


# ============================================================
# CONTRASTES FRENTE A CERO Y AL UMBRAL EMPRESARIAL
# ============================================================
conv = diferencia_wald(x_B, n_B, x_A, n_A)  ### Evalúa si la diferencia observada supera cero.

negocio = diferencia_wald(x_B, n_B, x_A, n_A, UMBRAL_NEGOCIO)  ### Evalúa si la diferencia supera el umbral empresarial de +1.0 pp.


# ============================================================
# PRESENTACIÓN DE LOS RESULTADOS DEL CONTRASTE
# ============================================================
print(f"z frente a cero: {conv['z']:.4f}")  ### Muestra el estadístico z del contraste frente a cero.

print(f"p unilateral frente a cero: {conv['p_mayor']:.6g}")  ### Muestra el p-valor para determinar si B supera a A.

print(f"Límite inferior unilateral 95%: {conv['li_uni95']*100:.3f} pp")  ### Muestra el límite inferior unilateral en puntos porcentuales.

print(f"IC bilateral 95%: [{conv['ic95_low']*100:.3f}, {conv['ic95_high']*100:.3f}] pp")  ### Presenta el intervalo bilateral para describir la incertidumbre del efecto.

print(f"p unilateral frente a +1,0 pp: {negocio['p_mayor']:.6g}")  ### Muestra el p-valor al contrastar directamente el umbral empresarial.

z frente a cero: 5.4436
p unilateral frente a cero: 2.61091e-08
Límite inferior unilateral 95%: 1.916 pp
IC bilateral 95%: [1.757, 3.733] pp
p unilateral frente a +1,0 pp: 0.000269553


### **Evidencia estadística y superación del umbral empresarial**

El estadístico **z = 5.4436** y el **p-valor = 2.61091e-08** muestran evidencia estadística de que la conversión de **B es mayor que la de A**, ya que el p-valor es menor que **α = 0.05**.

El **límite inferior unilateral del 95 % es +1.916 pp**, por lo que la mejora mínima compatible con este intervalo se mantiene por encima de **0 pp**. Además, el **IC bilateral del 95 % es [1.757, 3.733] pp**, indicando un rango plausible para la diferencia de conversión B−A.

Al contrastar directamente el **umbral empresarial de +1.0 pp**, se obtiene un **p-valor = 0.000269553**, también menor que 0.05. Esto indica evidencia estadística de que la mejora observada **supera el umbral empresarial establecido**.

In [ ]:
# ============================================================
# PREPARACIÓN DE DATOS PARA EL GRÁFICO
# ============================================================
grafico = tabla.reset_index()  ### Convierte el índice de la tabla en una columna para utilizarlo en el gráfico.

grafico["conversion_pct"] = grafico["tasa_conversion"] * 100  ### Convierte la tasa decimal a porcentaje.


# ============================================================
# CREACIÓN DEL GRÁFICO DE CONVERSIÓN
# ============================================================
fig = px.bar(grafico, x="grupo", y="conversion_pct", text="conversion_pct",
             title="Conversión observada — datos simulados",
             labels={"conversion_pct": "Usuarios que compraron (%)", "grupo": "Grupo"})

fig.update_traces(texttemplate="%{text:.2f}%", textposition="outside")  ### Muestra el porcentaje encima de cada barra con dos decimales.

fig.update_yaxes(rangemode="tozero")  ### Hace que el eje vertical comience desde cero para facilitar la comparación.

fig.show()  ### Muestra el gráfico de conversión de los grupos A y B.


### **Comparación visual de las tasas de conversión**

El gráfico muestra que **A_Control** alcanza una conversión de **11.39 %**, mientras que **B_Tratamiento** alcanza **14.14 %**. Visualmente, B presenta una tasa de conversión mayor que A, consistente con la diferencia observada de **+2.745 puntos porcentuales**.

La diferencia entre las barras permite apreciar de forma directa la mejora observada en B. Sin embargo, este gráfico únicamente describe los resultados de la muestra y debe complementarse con el **contraste estadístico y los intervalos de confianza** para evaluar la evidencia estadística del efecto.

### **Análisis y conclusión estadística**

**1. Registre alfa, z y p unilateral frente a cero.**

El **alfa es 0.05**, el estadístico **z = 5.4436** y el **p-valor unilateral frente a cero = 2.61091e-08**.

**2. ¿Se rechaza H0? Explique en el contexto de conversión del caso simulado.**

Sí, **se rechaza H0**, porque el p-valor es menor que **α = 0.05**. En el contexto del experimento, existe evidencia estadística de que la conversión de **B es mayor que la de A**.

**3. ¿El límite inferior unilateral supera cero? ¿Y el umbral de 1,0 pp?**

Sí. El límite inferior unilateral del 95 % es **+1.916 pp**, por lo que supera **0 pp** y también supera el **umbral empresarial de +1.0 pp**.

**4. Explique por qué el IC bilateral del 95 % y el límite unilateral del 95 % tienen extremos distintos.**

El **IC bilateral del 95 %** y el **límite unilateral del 95 %** tienen extremos distintos porque utilizan diferentes distribuciones de probabilidad en las colas. El IC bilateral reparte el **5 % de error entre ambas colas**, mientras que el intervalo unilateral concentra ese **5 % en una sola cola**.

**5. ¿Superar 1,0 pp con la estimación puntual demuestra superar ese umbral con evidencia suficiente?**

No. Que la estimación puntual sea mayor que **1.0 pp** no demuestra por sí sola que se supere el umbral con evidencia suficiente. Es necesario contrastar directamente el **umbral de +1.0 pp** y considerar la incertidumbre de la estimación.


## Actividad 6 — Del resultado estadístico a la decisión
### Paso 8. Evaluar el guardrail con incertidumbre
El guardrail es la proporción de usuarios con alguna incidencia de pago en siete días. El margen de deterioro admisible es +0,5 pp. La no inferioridad exige que el límite superior unilateral del 95 % esté por debajo de ese margen. No basta obtener p > 0,05 en una prueba de igualdad.


In [15]:
# ============================================================
# ANÁLISIS DEL GUARDRAIL
# ============================================================

# Obtener las incidencias de cada grupo
guardrail = ab.groupby("grupo")["incidencia_pago_7d"].agg(["sum", "count", "mean"])
guardrail.columns = ["incidencias", "n", "tasa_incidencia"]  ### Resume las incidencias y calcula la tasa de incidencia por grupo.

## Obtener las incidencias de cada grupo
g_A = int(guardrail.loc["A_Control", "incidencias"])  ### Extrae las incidencias observadas en A.
g_B = int(guardrail.loc["B_Tratamiento", "incidencias"])  ### Extrae las incidencias observadas en B.

## Contrastar el guardrail frente al margen permitido
guard = diferencia_wald(g_B, n_B, g_A, n_A, MARGEN_GUARDRAIL)  ### Evalúa el deterioro frente al margen permitido de +0.5 pp.

## Mostrar los resultados
display(guardrail)  ### Presenta las incidencias y tasas observadas en cada grupo.
print(f"Cambio del guardrail B−A: {guard['d']*100:.3f} pp")  ### Muestra el cambio observado entre B y A.
print(f"Límite superior unilateral 95%: {guard['ls_uni95']*100:.3f} pp")  ### Muestra el límite superior para evaluar la no inferioridad.
print(f"Margen permitido: {MARGEN_GUARDRAIL*100:.1f} pp")  ### Muestra el deterioro máximo permitido.
print(f"p de no inferioridad: {guard['p_menor']:.6g}")  ### Muestra el p-valor del contraste de no inferioridad.


,incidencias,n,tasa_incidencia
grupo,,,
A_Control,139,8743,0.015898
B_Tratamiento,176,8743,0.020130


Cambio del guardrail B−A: 0.423 pp
Límite superior unilateral 95%: 0.754 pp
Margen permitido: 0.5 pp
p de no inferioridad: 0.351286


### **Evaluación del guardrail frente al margen permitido**

La tasa de incidencia fue de **1.590 % en A_Control** y **2.013 % en B_Tratamiento**, generando un aumento observado de **+0.423 puntos porcentuales**.

Aunque el aumento observado de **0.423 pp** es menor que el margen permitido de **+0.5 pp**, el **límite superior unilateral del 95 % es +0.754 pp**, que supera el margen establecido. Por lo tanto, **no se demuestra la no inferioridad del guardrail** bajo el criterio definido.

El **p de no inferioridad = 0.351286** tampoco permite demostrar que el deterioro se encuentre dentro del margen permitido. Por ello, el guardrail **no queda protegido estadísticamente**, aunque el cambio observado puntual sea menor que +0.5 pp.

### Paso 9. Integrar la regla de decisión
| Criterio | Evidencia requerida |
|---|---|
| Diseño y calidad | Una fila por usuario, asignación persistente, muestra y seguimiento completos |
| Mejora frente a cero | p unilateral < 0,05 |
| Relevancia empresarial | Límite inferior unilateral del 95 % > +1,0 pp |
| Guardrail | Límite superior unilateral del 95 % < +0,5 pp |

La regla se fijó antes de simular. El MDE dimensiona la muestra y no reemplaza al umbral empresarial. Si no se satisface un requisito, distinguir evidencia insuficiente de evidencia de perjuicio. Cualquier experimento posterior necesita un nuevo plan; no seguir agregando datos hasta lograr significancia.


In [16]:
# ============================================================
# EVALUACIÓN DE LOS CRITERIOS DE DECISIÓN
# ============================================================

# Evaluar cada criterio de la regla de decisión
cumple_estadistica = bool(conv["p_mayor"] < ALPHA)  ### Verifica si la mejora frente a cero es estadísticamente significativa.

cumple_negocio = bool(conv["li_uni95"] > UMBRAL_NEGOCIO)  ### Verifica si el límite inferior supera el umbral empresarial de +1.0 pp.

cumple_guardrail = bool(guard["ls_uni95"] < MARGEN_GUARDRAIL)  ### Verifica si el límite superior del guardrail está por debajo del margen permitido.

cumple_diseno = bool((tam_grupos >= N_POR_GRUPO).all())  ### Verifica que ambos grupos cumplan con la muestra planificada.

## Integrar los criterios en una sola estructura
criterios = pd.Series({"Diseño validado": cumple_diseno,
                       "Mejora frente a cero": cumple_estadistica,
                       "Mejora superior al umbral": cumple_negocio,
                       "Guardrail protegido": cumple_guardrail})  ### Reúne los cuatro criterios definidos antes de la simulación.

display(criterios.to_frame("cumple"))  ### Muestra si cada criterio cumple o no cumple.

## Aplicar la regla conjunta de decisión
if criterios.all():
    decision = "Recomendar B en el caso simulado, con seguimiento posterior."  ### Solo se recomienda B si todos los criterios se cumplen.
else:
    decision = "No recomendar el despliegue todavía: revisar los criterios no cumplidos."  ### Si algún criterio falla, no se recomienda el despliegue.

print(decision)  ### Muestra la decisión resultante de la regla conjunta.

print("Conclusión académica con datos simulados; no valida un despliegue real.")  ### Aclara que la conclusión corresponde al ejercicio académico.

,cumple
Diseño validado,True
Mejora frente a cero,True
Mejora superior al umbral,True
Guardrail protegido,False


No recomendar el despliegue todavía: revisar los criterios no cumplidos.
Conclusión académica con datos simulados; no valida un despliegue real.


### **Integración de los criterios de decisión**

El resultado muestra que el **diseño está validado**, la **mejora frente a cero cumple** y la **mejora supera el umbral empresarial**. Sin embargo, el criterio de **guardrail protegido no cumple**, porque el límite superior unilateral del 95 % fue **+0.754 pp**, superior al margen permitido de **+0.5 pp**.

Por lo tanto, al no cumplirse todos los criterios de la regla conjunta, la conclusión del experimento es **no recomendar el despliegue todavía y revisar el criterio no cumplido**. Esta conclusión corresponde únicamente a los **datos simulados del ejercicio académico**.

### **Pausa de decisión**

| Componente | Resultado observado | Interpretación |
|---|---|---|
| p unilateral frente a cero | **2.61091e-08** | Es menor que α = 0.05, por lo que existe evidencia estadística de que B supera a A en conversión. |
| Diferencia B−A en pp y lift relativo | **+2.745 pp y +24.10 %** | B presenta una mejora observada de 2.745 puntos porcentuales, equivalente a un lift relativo de 24.10 %. |
| Límite inferior y umbral empresarial | **+1.916 pp > +1.0 pp** | El límite inferior supera el umbral empresarial, por lo que se cumple el criterio de relevancia empresarial. |
| Límite superior del guardrail y margen | **+0.754 pp > +0.5 pp** | El límite superior supera el margen permitido, por lo que no se demuestra la protección del guardrail. |
| Calidad y límites del caso simulado | **17 486 usuarios; 8743 por grupo** | El diseño y seguimiento cumplen lo planificado, pero los resultados corresponden a datos simulados y el guardrail no cumple. |

**Decisión en cuatro líneas:** **No recomendar todavía**, porque aunque la mejora frente a cero es estadísticamente significativa (**p = 2.61091e-08**) y el límite inferior alcanza **+1.916 pp**, el límite superior del guardrail es **+0.754 pp**, por encima del margen permitido de **+0.5 pp**. Si se requiere continuar, debe plantearse un **nuevo diseño válido** con criterios predefinidos, sin seguir agregando datos de esta misma muestra hasta obtener significancia.


## Actividad 7 — Retos de aplicación y retroalimentación

Se resuelven durante la clase y no generan una entrega adicional.

### Reto 1 — Estimación y umbral

Un caso presenta p = 0,02 frente a cero, mejora estimada de 0,4 pp y umbral empresarial de 3 pp. ¿Qué demuestra la prueba y qué no demuestra? ¿Qué información aporta el intervalo?

La prueba muestra evidencia estadística de que la diferencia es mayor que cero, porque **p = 0,02 < 0,05**. Sin embargo, no demuestra que la mejora alcance el **umbral empresarial de 3 pp**. El intervalo permite evaluar la incertidumbre de la estimación y determinar si la mejora compatible con los datos supera o no ese umbral.

### Reto 2 — Cambio de pregunta

Si interesa cualquier diferencia entre A y B, formule H0 y H1 e indique el tipo de contraste. ¿Por qué debe elegirse antes de observar los resultados?

**H0: pB − pA = 0** y **H1: pB − pA ≠ 0**. Corresponde a un **contraste bilateral**, porque interesa detectar diferencias tanto positivas como negativas. Debe definirse antes de observar los resultados para evitar adaptar la hipótesis o el tipo de contraste a los datos obtenidos.

### Reto 3 — Guardrail no concluyente

El margen es +0,5 pp. El aumento estimado de incidencias es +0,2 pp, pero el límite superior unilateral es +0,7 pp. ¿Se ha demostrado cumplir el guardrail? ¿Es correcto afirmar que se ha demostrado perjuicio?

No se ha demostrado cumplir el guardrail, porque el **límite superior de +0,7 pp** supera el margen permitido de **+0,5 pp**. Tampoco es correcto afirmar que se ha demostrado perjuicio, porque el resultado indica **evidencia insuficiente para demostrar la no inferioridad**, no una demostración de que exista un deterioro real.

### Reto 4 — Reproducibilidad y sesgo

Un equipo cambia la semilla hasta conseguir una decisión favorable. Explique por qué eso invalida el ejercicio. Proponga cómo documentar una exploración de escenarios sin sustituir el caso base.

Cambiar la semilla hasta obtener un resultado favorable introduce un **sesgo de selección del resultado** y afecta la reproducibilidad del experimento. La semilla debe fijarse antes de analizar los resultados. Para explorar escenarios, se pueden documentar semillas o simulaciones adicionales por separado, indicando que son análisis de sensibilidad y manteniendo el **caso base original** como referencia.

# VIII. Secuencia de trabajo para la sesión
| Momento | Minutos | Desarrollo |
|---|---:|---|
| Apertura | 10 | Pregunta orientadora y caso |
| Actividad 1 | 15 | Activación conceptual |
| Actividad 2 | 20 | Representación e hipótesis |
| Actividad 3 | 30 | Planificación, simulación y calidad |
| Receso | 15 | Pausa |
| Actividad 4 | 25 | Sensibilidad de muestra |
| Actividad 5 | 30 | Tasas, contraste e intervalos |
| Actividad 6 | 20 | Guardrail y decisión |
| Actividad 7 | 15 | Retos y ticket de salida |

Total: 180 minutos, incluido el receso.

## Ticket de salida
«Un experimento A/B no termina en el p-valor porque…».


# IX. Lista de verificación de aprendizaje
- [ x ] Identifico factor, niveles y unidad experimental.
- [ x ] Defino respuesta, denominador y ventana de observación.
- [ x  ] Formulo hipótesis antes de observar resultados.
- [ x ] Distingo baseline, MDE, umbral empresarial, alfa y potencia.
- [ x ] Calculo muestra antes de generar los datos del caso.
- [ x ] Interpreto pp, lift, p-valor e intervalos sin confundirlos.
- [ x ] Evalúo el guardrail con margen e incertidumbre.
- [ x  ] Distingo evidencia insuficiente de evidencia de perjuicio.
- [ x ] Reconozco peeking, contaminación y selección de resultados.
- [ x ] Declaro el carácter simulado y los límites del ejercicio.


# X. Referencias
- Kohavi, R., Tang, D., y Xu, Y. (2020). Trustworthy Online Controlled Experiments. Cambridge University Press.
- Bruce, P., Bruce, A., y Gedeck, P. (2020). Practical Statistics for Data Scientists, segunda edición. O’Reilly.
- statsmodels. Documentación de NormalIndPower y proportion_effectsize: https://www.statsmodels.org/stable/stats.html
- SciPy. Distribución normal, funciones cdf, sf y ppf: https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.norm.html

Datos: simulación académica NexaCommerce incluida en el notebook, semilla 20260921. Los datos no proceden de las referencias bibliográficas.
